### PR TITLE
Feture/services

### PR DESCRIPTION
**Pull request by Nazar865**
add base implementation for CardService

### CODE REVIEW

File: Corzbank.Data/CorzbankDbContext.cs [hunk 1]
```diff
@@ -17,7 +17,51 @@ public CorzbankDbContext(DbContextOptions<CorzbankDbContext> options)
...
+                .HasOne(d => d.Deposit)
+                .WithMany(dc => dc.DepositCards)
+                .HasForeignKey(d => d.DepositId);
+
+            modelBuilder.Entity<DepositCard>()
+               .HasOne(d => d.Card)
+               .WithMany(dc => dc.DepositCards)
+               .HasForeignKey(c => c.CardId);
+
+
```
Comment (YuriyBen) [inline Corzbank.Data/CorzbankDbContext.cs:35]: odd lines

File: Corzbank.Data/CorzbankDbContext.cs [hunk 2]
```diff
@@ -17,7 +17,51 @@ public CorzbankDbContext(DbContextOptions<CorzbankDbContext> options)
...
+                .HasOne(sd => sd.SenderCard)
+                .WithMany(st => st.SendTransfers)
+                .HasForeignKey(z => z.SenderCardId);
+
+            modelBuilder.Entity<TransferCard>()
+                .HasOne(rc => rc.ReceiverCard)
+                .WithMany(rt => rt.ReceiveTransfers)
+                .HasForeignKey(rc => rc.ReceiverCardId)
+                .IsRequired(false);
+
```
Comment (YuriyBen) [inline Corzbank.Data/CorzbankDbContext.cs:52]: line

File: Corzbank.Api/Controllers/CardController.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Corzbank.Api.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class CardController : ControllerBase
```
Comment (ghost) [inline Corzbank.Api/Controllers/CardController.cs]: Rename to Card**S**

File: Corzbank.Api/Startup.cs [hunk 4]
```diff
@@ -29,9 +29,14 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddControllers();
+            services.AddControllers()
+  .AddNewtonsoftJson(options =>
+      options.SerializerSettings.ReferenceLoopHandling =
+        Newtonsoft.Json.ReferenceLoopHandling.Ignore
+   );
```
Comment (ghost) [inline Corzbank.Api/Startup.cs]: missed tabs. Pay attention to formatting always.

File: Corzbank.Helpers/GenerateCard.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace Corzbank.Helpers
+{
+    public static class CardGenerator
+    {
+        public static Card GenerateCard(this CardModel card)
+        {
+            Card card1 = new Card
```
Comment (ghost) [inline Corzbank.Helpers/GenerateCard.cs]: wrong variable name. Fix this. Just card. Without 1, 2, 3, ...

File: Corzbank.Helpers/GenerateCard.cs [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+            {
+                CardNumber = GenerateCardNumber(card.PaymentSystem),
```
Comment (Nazar865) [inline Corzbank.Helpers/GenerateCard.cs:16]: i'm doing this now

File: Corzbank.Api/Controllers/CardController.cs (hunk 3 above)
Comment (Nazar865) [reply Corzbank.Api/Controllers/CardController.cs]: resolved

File: Corzbank.Api/Startup.cs (hunk 4 above)
Comment (Nazar865) [reply Corzbank.Api/Startup.cs]: resolved

File: Corzbank.Helpers/GenerateCard.cs (hunk 5 above)
Comment (Nazar865) [reply Corzbank.Helpers/GenerateCard.cs]: resolved

**Pull request merged**